projects/ion/src/lib/smart-table/smart-table.component.html

```diff
@@ -82,7 +82,9 @@
       >
         <!-- cell types -->
 
-        <span class="td-span" *ngIf="!column.type">{{ row[column.key] }}</span>
+        <span class="td-span" *ngIf="!column.type">{{
+          row[column.key] || '-'
```

Comment:
This can be a problem when the data will be 0

Reply:
``
```suggestion
          row[column.key] ?? '-'
```

Reply:
```suggestion
          row[column.key] !== null && row[column.key] !== undefined && row[column.key] !== '' ? row[column.key] : '-'
```

So it checks that it's not null, undefined, or empty before deciding whether to display the value or the '-'. If a property has a valid value it will be displayed in the table, otherwise a hyphen will be displayed. This will allow you to display numeric values ​​equal to zero while also displaying a '-' on values ​​that are null, undefined or empty.

Reply:
I think we can create a directive for this, as we would avoid putting logic in the HTML. I make myself available to do/help with that. What do you think @iurynogueira ?

Reply:
Great idea @AlyssonMascarenhas this can be used in the future in other components.

Reply:
We can make this directive/pipe on another issue and change it here later? @wermesonsl and I need these changes to complete a task on SIGA project. 🥲

Reply:
This afternoon I will open editing and do the same